StreamChatUITestsAppUITests/Tests/Message Delivery Status/MessageDeliveryStatus+ChannelList_Tests.swift

```diff
@@ -0,0 +1,305 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+
+final class MessageDeliveryStatus_ChannelList_Tests: StreamTestCase {
+
+    let message = "message"
+    var failedMessage: String { "failed \(message)" }
+
+    let threadReply = "thread reply"
+    var pendingThreadReply: String { "pending \(threadReply)" }
+    var failedThreadReply: String { "failed \(threadReply)" }
+
+    func test_deliveryStatusClocksShownInPreview_whenTheLastMessageIsInPendingState() {
+        linkToScenario(withId: 166)
+
+        GIVEN("user opens the channel") {
+            userRobot
+                .login()
+                .openChannel()
+            backendRobot.delayServerResponse(byTimeInterval: 10.0)
+        }
+        AND("user sends new message") {
+            userRobot.sendMessage(message)
+        }
+        WHEN("user retuns to the channel list before the message is sent") {
+            userRobot.tapOnBackButton()
+        }
+        THEN("last message delivery status in the channel preview shows clocks on the left") {
+            userRobot
+                .assertMessageReadCountInChannelPreview(readBy: 0)
+                .assertMessageDeliveryStatusInChannelPreview(.pending)
+        }
+    }
+
+    func test_singleCheckmarkShownInPreview_whenTheLastMessageIsSent() {
+        linkToScenario(withId: 167)
+
+        GIVEN("user opens the channel") {
+            userRobot
+                .login()
+                .openChannel()
+            backendRobot.delayServerResponse(byTimeInterval: 10.0)
+        }
+        AND("user sends new message") {
+            userRobot.sendMessage(message)
+        }
+        AND("message is succesfully sent") {
+            userRobot.waitForNewMessage(withText: message)
+        }
+        WHEN("user retuns to the channel list") {
+            userRobot.tapOnBackButton()
+        }
+        THEN("last message delivery status in the channel preview shows clocks on the left") {
+            userRobot
+                .assertMessageReadCountInChannelPreview(readBy: 0)
+                .assertMessageDeliveryStatusInChannelPreview(.pending)
+
+        }
+    }
+
+    func test_errorIndicatorShownInPreview_whenMessageFailedToBeSent() {
+        linkToScenario(withId: 168)
+
+        GIVEN("user opens the channel") {
+            deviceRobot.setConnectivitySwitchVisibility(to: .on)
+            userRobot
+                .login()
+                .openChannel()
+        }
+        AND("user's message is not sent") {
+            deviceRobot.setConnectivity(to: .off)
+            userRobot
+                .sendMessage(failedMessage)
+                .assertMessageFailedToBeSent()
+        }
+        WHEN("user retuns to the channel list") {
+            userRobot.tapOnBackButton()
+        }
+        THEN("error indicator is shown for the failed message") {
+            userRobot
+            .assertMessageDeliveryStatus(.failed)
+            .assertMessageReadCount(readBy: 0)
```

Comment:
nit: smth is wrong with the formatting here. Why don't we use lint to UI tests, btw?